### PR TITLE
Add error message in _pushInternalMessage when unknown type

### LINKED ISSUE #3542: Assert that type is found when doing store.push
We should add an assert in `store._pushInternalModel` checking that `type` is found.

### PR DESCRIPTION
**Pull request by Serabe**
When pushing data with an unknown type, an error is raised.
Fix #3542

### CODE REVIEW

File: packages/ember-data/tests/unit/store/push-test.js [hunk 1]
```diff
@@ -476,6 +476,18 @@ test('Calling push with a link for a non async relationship should warn', functi
...
 });
 
+test('Calling push with an unknown model name throws an assertion error', function() {
+
+  expectAssertion(function() {
+    run(function() {
+      store.push({
+        id: '1',
+        type: 'unknown'
+      });
```
Comment (wecc) [inline packages/ember-data/tests/unit/store/push-test.js]: This should be valid JSON API and has to be wrapped in `data`:

```js
{
  data: {
    type: 'unknown',
    id: '1'
  }
}
```

**Comment by Serabe** [comment]
Thank you, @wecc. Solved.

**Comment by wecc** [comment]
LGTM, @bmac?

**Comment by bmac** [comment]
Looks good. @Serabe Could you amend the commit message to start with [BUGFIX release] so we can include this in a future 1.13 release?

**Comment by Serabe** [comment]
Done!

**Comment by bmac** [comment]
Thanks @Serabe

**Pull request merged**